frontend/public/components/app.jsx

```diff
@@ -167,6 +199,12 @@ class App extends React.PureComponent {
   }
 }
 
+const appStateToProps = (state) => ({
+  user: state.UI.get('user'),
+});
+
+const App = connect(appStateToProps, {})(connectToFlags(FLAGS.OPENSHIFT)(App_));
```

Comment:
I wonder if we shouldn't add this logic to the masthead instead. It already has the user and the OpenShift flag, and it already handles the logic for logout. Since we're adding the listeners to `window` it doesn't strictly need to be in the `App` component.

Reply:
Interestingly, I have a TamperMonkey script which deletes the masthead (because it is huge and unnecessary for my daily usage of the console). Obviously this is not supported, but would this mean I would never get logged out?

Reply:
Heh, maybe? :) I guess it depends on whether it's hiding it with CSS or blocking the code from running.

If someone were determined to get around inactivity timeout, they could. For instance, by using an extension that occasionally generates mouse clicks. There's nothing we can do to block that. Even with the CLI inactivity timeout, which is all server-side, you could have a cronjob that occasionally hit the API to keep the token fresh.

The tokens do have a lifetime (24 hours by default) that you can't get around. So it will still expire.